jax/_src/cudnn/fused_attention_stableHLO.py

```diff
@@ -0,0 +1,672 @@
+_dot_product_attention_bwd_p_wrapper.multiple_results = True
+_dot_product_attention_bwd_p_wrapper.def_impl(_dot_product_attention_bwd_impl)
+_dot_product_attention_bwd_p_wrapper.def_abstract_eval(_dot_product_attention_bwd_abstract)
+
+
+batching.primitive_batchers[_dot_product_attention_fwd_p_wrapper] = _dot_product_attention_fwd_batcher
+batching.primitive_batchers[_dot_product_attention_bwd_p_wrapper] = _dot_product_attention_bwd_batcher
+
+_dot_product_attention_fwd_lower.def_partition(
+    infer_sharding_from_operands=_dot_product_attention_fwd_infer_sharding_from_operands,
+    partition=_dot_product_attention_fwd_partition)
+
+mlir.register_lowering(_dot_product_attention_fwd_p_wrapper,
+                        mlir.lower_fun(_dot_product_attention_fwd_lower, multiple_results=True))
+                        
+_dot_product_attention_bwd_lower.def_partition(
+    infer_sharding_from_operands=_dot_product_attention_bwd_infer_sharding_from_operands,
+    partition=_dot_product_attention_bwd_partition)
+
+mlir.register_lowering(_dot_product_attention_bwd_p_wrapper,
+                        mlir.lower_fun(_dot_product_attention_bwd_lower, multiple_results=True))
+
+dispatch.prim_requires_devices_during_lowering.add(_dot_product_attention_fwd_p)
+dispatch.prim_requires_devices_during_lowering.add(_dot_product_attention_fwd_p_wrapper)
+dispatch.prim_requires_devices_during_lowering.add(_dot_product_attention_bwd_p)
+dispatch.prim_requires_devices_during_lowering.add(_dot_product_attention_bwd_p_wrapper)
+
+@partial(jax.custom_vjp, nondiff_argnums=(5, 6, 7, 8, 9, 10))
+def _dot_product_attention(query: Array,
+                            key: Array,
+                            value: Array,
+                            bias: Array,
+                            mask: Array,
+                            scale: float,
+                            seed: int,
+                            dropout_rate: float,
+                            variadic_args: tuple[bool],
+                            is_flash_attention: bool,
+                            is_causal_mask: bool):
+    output = _dot_product_attention_fwd( 
+        query, key, value, bias, mask,
+        scale=scale, seed=seed, dropout_rate=dropout_rate, variadic_args=variadic_args,
+        is_flash_attention=is_flash_attention, is_causal_mask=is_causal_mask)
+    return output
+
+# _dot_product_attention_fwd must have the same func signature as _dot_product_attention
+_dot_product_attention.defvjp(_dot_product_attention_fwd_rule, _dot_product_attention_bwd_rule)
+    
+# User interface
+def dot_product_attention(query: Array,
+                          key: Array,
+                          value: Array,
+                          scale: float = 1.0,
+                          bias: Optional[Array] = None,
+                          mask: Optional[Array] = None,
+                          is_causal_mask: bool = False,
+                          seed: int = 42,
```

Comment:
I'm wondering if there is an easy way to make this compatible with JAX PRNG key. @froystig any ideas?